src/tree/render.js

```diff
@@ -182,124 +142,7 @@ const renderTree = async ({ element, layout, app, model }) => {
-          x: function(d) {
-            d.xActual =
-              d.parent && d.parent.xActual
-                ? d.parent.xActual +
-                  nodeSize.width / 2 +
-                  siblingSpacing / 2 +
-                  (d.data.childNumber - d.parent.children.length / 2) * (nodeSize.width + siblingSpacing)
-                : 1;
-            return d.xActual;
-          },
-          y: function(d) {
-            return d.y;
-          },
-        },
-      };
-      break;
-    case 'btt':
-      orientations = {
-        'bottom-to-top': {
-          depthSpacing: -200,
-          pathOffsetSelf: {
-            x: nodeSize.width / 2,
-            y: nodeSize.height,
-          },
-          pathOffsetParent: {
-            x: nodeSize.width / 2,
-            y: 0,
-          },
-          x: function(d) {
-            d.xActual =
-              d.parent && d.parent.xActual
-                ? d.parent.xActual +
-                  nodeSize.width / 2 +
-                  siblingSpacing / 2 +
-                  (d.data.childNumber - d.parent.children.length / 2) * (nodeSize.width + siblingSpacing)
-                : 1;
-            return d.xActual;
-          },
-          y: function(d) {
-            return d.y;
-          },
-        },
-      };
-      break;
-    case 'ltr':
-      orientations = {
-        'left-to-right': {
-          depthSpacing: 500,
-          pathOffsetSelf: {
-            x: 0,
-            y: nodeSize.height / 2,
-          },
-          pathOffsetParent: {
-            x: nodeSize.width,
-            y: nodeSize.height / 2,
-          },
-          y: function(d) {
-            d.yActual =
-              d.parent && d.parent.yActual
-                ? d.parent.yActual +
-                  nodeSize.height / 2 +
-                  siblingSpacing / 2 +
-                  (d.data.childNumber - d.parent.children.length / 2) * (nodeSize.height + siblingSpacing)
-                : 1;
-            return d.yActual;
-          },
-          x: function(d) {
-            return d.y;
-          },
-        },
-      };
-      break;
-    case 'rtl':
-      orientations = {
-        'right-to-left': {
-          depthSpacing: -500,
-          pathOffsetSelf: {
-            x: nodeSize.width,
-            y: nodeSize.height / 2,
-          },
-          pathOffsetParent: {
-            x: 0,
-            y: nodeSize.height / 2,
-          },
-          y: function(d) {
-            d.yActual =
-              d.parent && d.parent.yActual
-                ? d.parent.yActual +
-                  nodeSize.height / 2 +
-                  siblingSpacing / 2 +
-                  (d.data.childNumber - d.parent.children.length / 2) * (nodeSize.height + siblingSpacing)
-                : 1;
-            return d.yActual;
-          },
-          x: function(d) {
-            return d.y;
-          },
-        },
-      };
-      break;
-    default:
-      break;
-  }
+  const orientations = position(orientation, nodeSize);
```

Comment:
The naming here is a bit weird. I don't know if we still want call it `orientations`, since it will only contain one orientation the way I refactored it. At least, I think this line should be something like `someName = getSomeName(...)`.

Reply:
yea maybe, I renamed these quite a few times before just picking one. I liked the simplicity of position, path, box... orientation didn't make much sense as it is position data you get back.
For "position" you can read it as a verb, "to position something", which makes more sense. but then, "to box something".... anyway, can always change it later, important part was to separate them.

Reply:
Ok, yeah better naming tend to pop up along the way :)